spec/webauthn/authenticator_assertion_response_spec.rb

```diff
@@ -3,30 +3,49 @@
 require "webauthn/authenticator_assertion_response"
 
 RSpec.describe WebAuthn::AuthenticatorAssertionResponse do
-  it "is invalid if type is not get" do
+  let(:challenge) { fake_challenge }
+  let(:encoded_challenge) { WebAuthn::Utils.ua_encode(challenge) }
+
+  it "is valid if everything's in place" do
     assertion_response = WebAuthn::AuthenticatorAssertionResponse.new(
-      client_data_json: hash_to_encoded_json(type: "webauthn.create"),
+      client_data_json: encoded_fake_client_data_json(challenge: challenge, type: "webauthn.get"),
       authenticator_data: fake_authenticator_data
     )
 
-    expect(assertion_response.valid?).to be_falsy
+    expect(assertion_response.valid?(encoded_challenge)).to be_truthy
   end
 
-  it "is invalid if user-present flag is off" do
-    assertion_response = WebAuthn::AuthenticatorAssertionResponse.new(
-      client_data_json: hash_to_encoded_json(type: "webauthn.get"),
-      authenticator_data: fake_authenticator_data(user_present: false)
-    )
+  describe "type validation" do
+    it "is invalid if type is not get" do
+      assertion_response = WebAuthn::AuthenticatorAssertionResponse.new(
+        client_data_json: encoded_fake_client_data_json(challenge: challenge, type: "webauthn.create"),
+        authenticator_data: fake_authenticator_data
+      )
+
+      expect(assertion_response.valid?(encoded_challenge)).to be_falsy
+    end
+
+    it "is invalid if user-present flag is off" do
```

Comment:
User present flag validation ended-up nested inside "type validation" `describe`. Was that intentional?

Reply:
Nop! My bad 😆